operator/builtin/parser/xml/xml_test.go

```diff
@@ -0,0 +1,146 @@
+package xml
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/observiq/stanza/entry"
+	"github.com/observiq/stanza/operator"
+	"github.com/observiq/stanza/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+func TestParse(t *testing.T) {
+	testCases := []struct {
+		name           string
+		value          interface{}
+		expectedResult interface{}
+		expectedErr    error
+	}{
+		{
+			name:        "Empty string",
+			value:       "",
+			expectedErr: errors.New("failed to decode as xml"),
+		},
+		{
+			name:        "No elements",
+			value:       "regular string",
```

Comment:
I'm a little confused why this doesn't trigger the 'failed to decode as xml' failure case, do you understand that?

Reply:
I also expected that behavior, but we discovered in our unit tests that this doesn't happen. It looks like the decoder is not discriminate enough and allows for just raw char data. We did however check the spec for XML and there should always be a root element. So I think this failure is warranted on our end.